accepted/0022-git-buildpack.md

```diff
@@ -0,0 +1,64 @@
+# Git Support
+
+## Proposal
+
+Introduce an (optional) Git buildpack into each language buildpack family. The buildpack
+will be charged with providing `git` metadata as build-time and run-time environment
+variables.
+
+## Motivation
+
+Applications of any language may have a need for information extracted from
+`git` metadata. For example, the commit sha might be used as versioning
+information when building a Go binary or used when running a Rails application.
+
+## Implementation
+
+A new `git` buildpack will be developed to detect whether the `.git` directory and
+`BP_GIT_LOAD` environment variable is present and set to `true`, and then
+read the directory to extract the following environment variables so that they
+can be included in the built image.
+
+- `GIT_COMMIT_SHA`
+
+### Detection Criteria
+
+The buildpack will detect if:
+1. The `.git` directory is present
+2. The `BP_GIT_LOAD` environment variable is set to `true`.
+
+### Build Process
+
+The build process of the buildpack will find the values for the
+expected environment variables and add them to a layer for use at launch.
+
+### Buildpack Order Grouping
+
+The `git` buildpack can be added to each family buildpack or it can be added
+separately at the beginning of each ordering defined in a builder.
+
+For example:
+
+```toml
+[[order]]
+
+  [[order.group]]
+    id = "paketo-buildpacks/go"
+    version = "<version>"
+
+  [[order.group]]
+    id = "paketo-buildpacks/git"
+    version = "<version>"
+    optional = true
```

Comment:
```suggestion
  [[order.group]]
    id = "paketo-buildpacks/git"
    version = "<version>"
    optional = true

  [[order.group]]
    id = "paketo-buildpacks/go"
    version = "<version>"
```

I think you'd want to swap the ordering here so that the Go buildpack can use the variables set during build time.